posts/2024/2024_08_22_inigo_final_report.rst

```diff
@@ -0,0 +1,335 @@
+
+  * After thorough discussion with my mentors, we decided that regularizing the latent space could be a good idea due to two reasons: to avoid "white spaces" in the latent space it that could lead undefined points due to the possible discontinuity of the space; and to prevent the model from learning the identity function, what is a known issue with unregularized AutoEncoders, though it does not always happen. The only addition to the original AE architecture was the inclusion of the Kullback-Leibler divergence loss term in the loss function, and two dense layers after the convolutional layers of the encoder, to output the :math:`\mu` and :math:`\sigma^2` of the latent space vector :math:`z`.
+
+  * The model is found in the ``vae_model.py`` module of the `TractoEncoder GSoC <https://github.com/itellaetxe/tractoencoder_gsoc>`_ repository. The architecture can be summarized in the figure below:
+
+  .. image:: /_static/images/gsoc/2024/inigo/inigo_variational_autoencoder.png
+    :class: custom-gsoc-margin
+    :alt: VAE architecture diagram.
+    :align: center
+    :width: 600
+
+  * It was necessary to implement Batch Normalization after the Encoder convolutional layers and exponential operator clipping, to prevent gradient explosion while training the model. The figure above displays that the encoder outputs the variance of the latent space vector (:math:`\sigma^2`), though it is a common practice to output the log-variance instead. This leads to using :math:`\sigma = e^{\frac{1}{2}\text{log}{\sigma^2}}`, which can get numerically quite unstable as it can shoot up very quickly when a sufficiently big log-variance is input. This modification allowed stable training and it was a major contribution to the robustness of the architecture. Shout out to my lab colleague `Jorge <https://github.com/jgarciacondado>`_ for the ideas and discussions on the topic.
+  * Weighing the Kullback-Leibler loss component was also implemented, based on the `Beta-VAE <https://openreview.net/forum?id=Sy2fzU9gl>`_ work, aiming for a stronger disentanglement of the latent space. However, this parameter was never explored (it was always set to 1.0) due to its trade-off with the reconstruction accuracy, but it is a potential improvement for future work in the context of hyperparameter optimization.
+  * The figure below shows a summary of the VAE results, also using the FiberCup dataset:
+
+    .. figure:: /_static/images/gsoc/2024/inigo/vanilla_vae_120_epoch_results.png
+      :class: custom-gsoc-margin
+      :alt: VAE architecture results on the FiberCup dataset.
+      :align: center
+      :width: 600
+
+      The bottom row shows two sets of unseen plausible streamlines, run through the model (encode & decode). We can see that the reconstruction fidelity is not as good as the vanilla AE, but it is still acceptable, considering that the model was only trained for 120 epochs, which took around 2 hours in my GPU-less laptop.
```

Comment:
I think this needs to change a bit because the figure changed

Reply:
true, thanks